omp/test/solver/batch_cg_kernels.cpp

```diff
@@ -0,0 +1,504 @@
+            ASSERT_LE(res_log_array[i * this->nrhs + j],
+                      this->opts_m.abs_residual_tol);
+
+            ASSERT_NEAR(
+                res_log_array[i * this->nrhs + j],
+                this->r_m.resnorm->get_const_values()[i * this->nrhs + j],
+                10 * r<value_type>::value);
+        }
+    }
+}
+
+
+TYPED_TEST(BatchCg, CoreSolvesSystemJacobi)
+{
+    using value_type = typename TestFixture::value_type;
+    using Mtx = typename TestFixture::Mtx;
+    using BDense = typename TestFixture::BDense;
+    using Solver = gko::solver::BatchCg<value_type>;
+    using LinSys = typename TestFixture::LinSys;
+    auto useexec = this->ompexec;
+    std::unique_ptr<typename Solver::Factory> batchcg_factory =
+        Solver::build()
+            .with_max_iterations(100)
+            .with_rel_residual_tol(1e-6f)
+            .with_preconditioner(gko::preconditioner::batch::type::jacobi)
+            .with_tolerance_type(gko::stop::batch::ToleranceType::relative)
+            .on(useexec);
+    const int nrhs_1 = 1;
+    const size_t nbatch = 3;
+    const LinSys sys = this->get_poisson_problem(nrhs_1, nbatch);
+    auto rx =
+        gko::batch_initialize<BDense>(nbatch, {0.0, 0.0, 0.0}, this->exec);
+    std::unique_ptr<Mtx> mtx = Mtx::create(useexec);
+    auto b = BDense::create(useexec);
+    auto x = BDense::create(useexec);
+    mtx->copy_from(gko::lend(sys.mtx));
+    b->copy_from(gko::lend(sys.b));
+    x->copy_from(gko::lend(rx));
+
+    std::unique_ptr<Solver> solver = batchcg_factory->generate(gko::give(mtx));
+    solver->apply(b.get(), x.get());
+
+    rx->copy_from(gko::lend(x));
+
+    GKO_ASSERT_BATCH_MTX_NEAR(rx, sys.xex, 1e-5 /*r<value_type>::value*/);
+}
+
+
+TYPED_TEST(BatchCg, UnitScalingDoesNotChangeResult)
+{
+    using Result = typename TestFixture::Result;
+    using BDense = typename TestFixture::BDense;
+    auto left_scale = gko::batch_initialize<BDense>(
+        this->nbatch, {1.0, 1.0, 1.0}, this->exec);
+    auto right_scale = gko::batch_initialize<BDense>(
+        this->nbatch, {1.0, 1.0, 1.0}, this->exec);
+
+    Result result =
+        this->solve_poisson_uniform_1(left_scale.get(), right_scale.get());
+
+    GKO_ASSERT_BATCH_MTX_NEAR(result.x, this->sys_1.xex,
+                              1e-6 /*r<value_type>::value*/);
+}
+
+
+TYPED_TEST(BatchCg, GeneralScalingDoesNotChangeResult)
+{
+    using Result = typename TestFixture::Result;
+    using BDense = typename TestFixture::BDense;
+    auto left_scale = gko::batch_initialize<BDense>(
+        this->nbatch, {0.8, 0.9, 0.95}, this->exec);
+    auto right_scale = gko::batch_initialize<BDense>(
+        this->nbatch, {1.0, 1.5, 1.05}, this->exec);
+
+    Result result =
+        this->solve_poisson_uniform_1(left_scale.get(), right_scale.get());
+
+    GKO_ASSERT_BATCH_MTX_NEAR(result.x, this->sys_1.xex,
+                              1e-3 /*r<value_type>::value*/);
```

Comment:
If this is the best tolerance we can use, it's concerning.

Reply:
I guess 1e3 * eps is the best tol, we could use here.